### PR TITLE
Re-run all tests when Dockerfile or Github worflow files change

### LINKED ISSUE #8921: Add some "sensitive file" to include in "dock-only CI pytestless builds
**Description**

Currently we skip tests when no .py files change. But this does not take into account much rarer situations when Dockerfile.ci, setup.cfg and few other files change. 

We should add those files to the detection of whether to run the builds or not.

**Use case / motivation**

Sometimes we need to fix Dockerfile.ci for example and it woudl be great to see if the tests are still fine.

### PR DESCRIPTION
**Pull request by ashb**
Fixes #8921

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
FYI. @ashb It's enough to set VERBOSE_COMMANDS="true" in the workflow and all commands will be printed.

**Comment by potiuk** [comment]
The problem is that there are two greps - one displays and second counts:

```
Changed files matching the \.py$|.github/workflows/|^Dockerfile pattern

.github/workflows/ci.yml


Count changed files matching the \.py$|.github/workflows/|^Dockerfile pattern

0
```

**Comment by ashb** [comment]
Ah, thanks :)

File: scripts/ci/ci_count_changed_files.sh [hunk 1]
```diff
@@ -42,7 +42,7 @@ echo
 echo
 echo "Changed files matching the ${2} pattern"
 echo
-echo "${CHANGED_FILES}" | grep "${2}" || true
+echo "${CHANGED_FILES}" | grep -E "${2}" || true
 echo
 
 echo
```
Comment (potiuk) [inline scripts/ci/ci_count_changed_files.sh:48]: You also need to add -E to the grep below which actually counts the files.

Comment (ashb) [reply scripts/ci/ci_count_changed_files.sh:48]: > FYI. @ashb It's enough to set VERBOSE_COMMANDS="true" in the workflow and all commands will be printed.

Oh yes, I thought this was deeper in the call stack, so I thought that would have given me a lot more logs than I want. (Plus sometimes the more direct option is easier to remember)

**Comment by ashb** [comment]
I'm going to merge this, then my _next_ fix to the Dockerfile (to add  back in JAVA_HOME) will actually run the tests.

**Pull request merged**